# Исследование надежности заемщиков

Это первая часть проекта, она будет проверена автоматически. Вторую часть проверит ревьюер. 

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

Подключил необходимую библиотеку.

In [2]:
data = pd.read_csv('/datasets/data.csv')

Загрузил файл с базой данных и сохранил ее в переменную data.

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [34]:
display(data.head(20))

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   

       family_status  family_status_id gender income_type  debt  total_income  \
0    женат / замужем                 0      F   сотрудник     0        253875   
1    женат / замужем                 0      F   сотрудник     0        112080   
2    женат / замужем                 0      M   сотрудник     0        145885   
3    женат / замужем                 0      M   сотрудник     0        267628   
4   гражданский брак                 1      F   пенсионер     0        158616   
5   гражданский брак                 1      M   компаньон     0        255763   
6    женат / замужем                 0      F   компаньон     0        240525   
7    женат / замужем                 0      M   сотрудник     0        135823   
8   гражданский брак                 1      F   сотрудник     0         95856   
9    женат / замужем                 0      M   сотрудник     0        144425   
10   женат / замужем                 0      M   компаньон     0        113943   
11   женат / замужем                 0      F   сотрудник     0         77069   
12  гражданский брак                 1      M   пенсионер     0        118514   
13   женат / замужем                 0      F   сотрудник     0        130458   
14  гражданский брак                 1      F   компаньон     1        165127   
15   женат / замужем                 0      F   сотрудник     0        116820   
16   женат / замужем                 0      F   сотрудник     0        289202   
17  гражданский брак                 1      M   сотрудник     0         90410   
18    вдовец / вдова                 2      F   пенсионер     0         56823   
19         в разводе                 3      F   сотрудник     0        242831   

                                   purpose total_income_category  \
0                            покупка жилья                     B   
1                  приобретение автомобиля                     C   
2                            покупка жилья                     C   
3               дополнительное образование                     B   
4                          сыграть свадьбу                     C   
5                            покупка жилья                     B   
6                        операции с жильем                     B   
7                              образование                     C   
8                    на проведение свадьбы                     C   
9   

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [35]:
df = data.groupby('income_type')['total_income'].transform('median')
data['total_income'] = data['total_income'].fillna(df) 
display(data.head(30))

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   
20         1    1311.604166         36              среднее             1   
21         1     253.685166         33              среднее             1   
22         1    1766.644138         24              среднее             1   
23         0     272.981385         21               высшее             0   
24         1  338551.952911         57              среднее             1   
25         0  363548.489348         67              среднее             1   
26         0    2689.137274         41              среднее             1   
27         0     529.191635         28               высшее             0   
28         1     717.274324         26               высшее             0   
29         0  365286.622650         63              среднее             1   

            family_status  family_status_id gender  income_type  debt  \
0         женат / замужем                 0      F    сотрудник     0   
1         женат / замужем                 0      F    сотрудник     0   
2         женат / замужем                 0      M    сотрудник     0   
3         женат / замужем                 0      M    сотрудник     0   
4        гражданский брак                 1      F    пенсионер     0   
5        гражданский брак                 1      M    компаньон     0   
6         женат / замужем                 0      F    компаньон     0   
7         женат / замужем                 0      M    сотрудник     0   
8        гражданский брак                 1      F    сотрудник     0   
9         женат / замужем                 0      M    сотрудник     0   
10        женат / замужем                 0      M    компаньон     0   
11        женат / замужем                 0      F    сотрудник     0   
12       гражданский брак                 1      M    пенсионер     0   
13        женат / замужем                 0      F    сотрудник     0   
14       гражданский брак                 1      F    компаньон     1   
15        женат / замужем                 0      F    сотрудник     0   
16        женат / замужем                 0      F    сотрудник     0   
17       гражданский брак                 1      M    сотрудник     0   
18         вдовец / вдова                 2      F    пенсионер     0   
19              в разводе                 3      F    сотрудник     0   
20        женат / замужем                 0      M    сотрудник     0   
21       г

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [7]:
data['days_employed'] = abs(data['days_employed'])
print(data['days_employed'].head(10))

0      8437.673028
1      4024.803754
2      5623.422610
3      4124.747207
4    340266.072047
5       926.185831
6      2879.202052
7       152.779569
8      6929.865299
9      2188.756445
Name: days_employed, dtype: float64


**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [8]:
df = data.groupby('income_type')['days_employed'].median()
print(df)

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64


У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
df = data['children'].unique()
print(df)

[ 1  0  3  2 -1  4 20  5]


**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data.loc[(data['children'] != -1) & (data['children'] != 20)]
print(data.head(10))

   children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   Среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0     926.185831         27    высшее             0   
6         0    2879.202052         43    высшее             0   
7         0     152.779569         50   СРЕДНЕЕ             1   
8         2    6929.865299         35    ВЫСШЕЕ             0   
9         0    2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0    

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
df = data['children'].unique()
print(df)

[1 0 3 2 4 5]


### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждому типу занятости `income_type`.**

In [12]:
df = data.groupby('income_type')['days_employed'].transform('median')
data['days_employed'] = data['days_employed'].fillna(df)
print(data.head(30))

    children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              Среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              СРЕДНЕЕ             1   
8          2    6929.865299         35               ВЫСШЕЕ             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype('int')
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21402 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21402 non-null  int64  
 1   days_employed     21402 non-null  float64
 2   dob_years         21402 non-null  int64  
 3   education         21402 non-null  object 
 4   education_id      21402 non-null  int64  
 5   family_status     21402 non-null  object 
 6   family_status_id  21402 non-null  int64  
 7   gender            21402 non-null  object 
 8   income_type       21402 non-null  object 
 9   debt              21402 non-null  int64  
 10  total_income      21402 non-null  int32  
 11  purpose           21402 non-null  object 
dtypes: float64(1), int32(1), int64(5), object(5)
memory usage: 2.0+ MB


### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [15]:
data['education'] = data['education'].str.lower() 

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()
data.duplicated().sum()

0

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(messages):
    if 0 <= messages <= 30000:
        return 'E'
    if 30001 <= messages <= 50000:
        return 'D'
    if 50001 <= messages <= 200000:
        return 'C'
    if 200001 <= messages <= 1000000:
        return 'B'
    if messages >= 1000001:
        return 'A'

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)
print(data)

       children  days_employed  dob_years education  education_id  \
0             1    8437.673028         42    высшее             0   
1             1    4024.803754         36   среднее             1   
2             0    5623.422610         33   среднее             1   
3             3    4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1    4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1    2113.346888         38   среднее             1   
21523         3    3112.481705         38   среднее             1   
21524         2    1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем        

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
df = data['purpose'].unique()
print(df)

['покупка жилья' 'приобретение автомобиля' 'дополнительное образование'
 'сыграть свадьбу' 'операции с жильем' 'образование'
 'на проведение свадьбы' 'покупка жилья для семьи' 'покупка недвижимости'
 'покупка коммерческой недвижимости' 'покупка жилой недвижимости'
 'строительство собственной недвижимости' 'недвижимость'
 'строительство недвижимости' 'на покупку подержанного автомобиля'
 'на покупку своего автомобиля' 'операции с коммерческой недвижимостью'
 'строительство жилой недвижимости' 'жилье'
 'операции со своей недвижимостью' 'автомобили' 'заняться образованием'
 'сделка с подержанным автомобилем' 'получение образования' 'автомобиль'
 'свадьба' 'получение дополнительного образования' 'покупка своего жилья'
 'операции с недвижимостью' 'получение высшего образования'
 'свой автомобиль' 'сделка с автомобилем' 'профильное образование'
 'высшее образование' 'покупка жилья для сдачи' 'на покупку автомобиля'
 'ремонт жилью' 'заняться высшим образованием']


**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(i):
    if i == 'приобретение автомобиля' or i == 'авто' or i == 'на покупку подержанного автомобиля' or i == 'на покупку своего автомобиля' or i == 'автомобили' or i == 'сделка с подержанным автомобилем' or i == 'автомобиль' or i == 'свой автомобиль' or i == 'сделка с автомобилем' or i == 'на покупку автомобиля':
        return 'операции с автомобилем'
    if i == 'покупка жилья' or i == 'операции с жильем' or i == 'покупка жилья для семьи' or i == 'покупка недвижимости' or i == 'покупка коммерческой недвижимости' or i == 'покупка жилой недвижимости' or i == 'строительство собственной недвижимости' or i == 'недвижимость' or i == 'строительство недвижимости' or i == 'операции с коммерческой недвижимостью' or i == 'строительство жилой недвижимости' or i == 'жилье' or i == 'операции со своей недвижимостью' or i == 'покупка своего жилья' or i == 'операции с недвижимостью' or i == 'покупка жилья для сдачи' or i == 'ремонт жилью':
        return 'операции с недвижимостью'
    if i == 'дополнительное образование' or i == 'образование' or i == 'заняться образованием' or i == 'получение образования' or i == 'получение дополнительного образования' or i == 'получение высшего образования' or i == 'профильное образование' or i == 'высшее образование' or i == 'заняться высшим образованием':
        return 'получение образования'
    if i == 'сыграть свадьбу' or i == 'на проведение свадьбы' or i == 'свадьба':
        return 'проведение свадьбы'

In [22]:
print(categorize_purpose('авто'))

операции с автомобилем


In [23]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
print(data['purpose_category'])

0        операции с недвижимостью
1          операции с автомобилем
2        операции с недвижимостью
3           получение образования
4              проведение свадьбы
                   ...           
21520    операции с недвижимостью
21521      операции с автомобилем
21522    операции с недвижимостью
21523      операции с автомобилем
21524      операции с автомобилем
Name: purpose_category, Length: 21331, dtype: object


### Шаг 3. Исследуйте данные и ответьте на вопросы.

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [24]:
data_children = data.pivot_table(index='children', values='debt', aggfunc=['sum','count', 'mean']).round(3)

Построили сводную таблицу, определили количество семей у которых была просрочка по кредиту и общее количество семей.

In [25]:
data_children_mean = data_children['mean'] * 100
data_children['mean'] = data_children_mean

Высчитали долю просроченных кредитов по категориям в процентах

In [26]:
sum_debt = (data_children['sum']['debt'][3] + data_children['sum']['debt'][4] + data_children['sum']['debt'][5]) / 3
mean_debt = (data_children['mean']['debt'][3] + data_children['mean']['debt'][4] + data_children['mean']['debt'][5]) / 3
three = sum_debt / mean_debt
print(f'Средняя доля категории где три и более детей составляет {three:.2}''%')
display(data_children)

Средняя доля категории где три и более детей составляет 1.7%


sum  count mean
          debt   debt debt
children                  
0         1063  14091  7.5
1          444   4808  9.2
2          194   2052  9.5
3           27    330  8.2
4            4     41  9.8
5            0      9  0.0

Посчитали среднюю долю для категории три и более детей.

**Вывод:** В результате проведенного анализы мы выяснили: 
1. 7.5% - доля просроченных кредитов в семье без детей;
2. 9.2% - доля просроченных кредитов в семье с одним ребенком;
3. 9.5% - доля просроченных кредитов в семье с двумя детьми;
4. 1.7% - усредненная доля просроченных кредитов для семей с тремя и более детьми.

**В результате анализа мы частично подтвердили гипотезу и выявили что:**
1. чем меньше детей в семье, тем чаще берутся кредиты и тем меньше доля просрочек по кредитам;
2. для категории три и более детей необходимо провести дополнительный анализ, т.к. данных для анализа недостаточно.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [27]:
data_family_status = data.pivot_table(index='family_status', values='debt', aggfunc=['sum','count', 'mean']).round(3)

Построили сводную таблицу, определили общее количество просроченных платежей по категориям.

In [28]:
data_family_status_mean = data_family_status['mean'] * 100
data_family_status['mean'] = data_family_status_mean
display(data_family_status)

sum  count mean
                      debt   debt debt
family_status                         
Не женат / не замужем  273   2796  9.8
в разводе               84   1189  7.1
вдовец / вдова          63    951  6.6
гражданский брак       385   4134  9.3
женат / замужем        927  12261  7.6

Высчитали долю просроченных кредитов по категориям в процентах.

Вывод: В результате проведенного анализы мы выяснили:
Чаще всего берут кредиты люди относящиеся к категории женат / замужем и выплачивают кредиты в 92,43% случаев. Доля таких людей около 60%. Реже всех берут кредиты люди относящиеся к категории вдовец / вдова, но у этой категории самый высокий процент выплат, хотя доля таких людей не более 5% от общего кличества.

У клиентов находящихся в статусе Не женат / не замужем самая высокая вероятность стать должниками, т.к. доля просроченных кредитов у них самая высокая и составляет 9.8%;
У клиентов находящихся в статусе женат / замужем, вдовец / вдова и в разводе самая низкая вероятность стать должниками, их доли просроченных кредитов 7.6%, 6.6% и 7.1% соответственно;
В результате анализа мы подвердили гипотезу и выявили что:

семейное положение оказывает влияние на процент успешной выплаты по кредиту.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [29]:
data_dohod = data.pivot_table(index='total_income_category', values='debt', aggfunc=['sum','count', 'mean']).round(3)

Построил сводную таблицу, определили общее количество просроченных платежей по категориям.

In [30]:
data_dohod_mean = data_dohod['mean'] * 100
data_dohod['mean'] = data_dohod_mean
display(data_dohod)

sum  count mean
                       debt   debt debt
total_income_category                  
A                         2     25  8.0
B                       354   5014  7.1
C                      1353  15921  8.5
D                        21    349  6.0
E                         2     22  9.1

Высчитал долю просроченных кредитов по категориям в процентах.

**Вывод:** В результате проведенного анализы мы выяснили:  
1. Самый низкий процент просроченных кредитов у категории людей **B 7.1%** и **С 8.5%**.
2. Данных для полноценного анализа по остальным категориям недостаточно.

**В результате анализа мы частично подтвердили гипотезу и выявили что:**
1. Уровень дохода оказывает влияние на процент успешной выплаты по кредиту;
2. Для полноценного анализа необходимы дополнительные данные по категориям А, D, E.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
data_purpose = data.pivot_table(index='purpose_category', values='debt', aggfunc=['sum','count', 'mean']).round(3)

Построил сводную таблицу, определили общее количество просроченных платежей по категориям.

In [32]:
data_purpose_mean = data_purpose['mean'] * 100
data_purpose['mean'] = data_purpose_mean
display(data_purpose)

sum  count mean
                         debt   debt debt
purpose_category                         
операции с автомобилем    400   4279  9.3
операции с недвижимостью  780  10751  7.3
получение образования     369   3988  9.3
проведение свадьбы        183   2313  7.9

Высчитал долю просроченных кредитов по категориям в процентах.

**Вывод:** В результате проведенного анализы мы выяснили:  
1. Самый низкий процент просроченных кредитов 7.3% у категории **операции с недвижимостью**;
2. Самый высокий процент 9.3% просроченных кредитов у категории **операции с автомобилем	** и **получение образования**

**В результате анализа мы подтвердили гипотезу и выявили что:**
1. Кредиты на **операции с недвижимостью** выплачиваются лучшем чем остальные;
2. Доля кредитов на **проведение свадьбы** не большая, для более точной аналитики требуется собрать больше данных.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 
1. Ошибки при построении сводных таблиц или выгрузки из БД.
2. Данные не собирались.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
1. При использовании медианных значений строки с пропущенными данными не удаляются, это делает выборку более информативной.
2. Результаты по столбцам с пропущенными данными усредняются, это делает выборку типичной, но дает возможность продолжить работать с остальной таблицей в полном объеме.

### Шаг 4: Общий вывод.

Было проведено исследование, чтобы выявить влияние факторов влияющих на факт погашения кредита в срок. Входные данные – статистика о платежеспособности клиентов. Результаты исследования будут учтены при построении модели кредитного скоринга – специальной системы которая оценивает способность потенциальных заемщиков вернуть кредит банку.

Перед проведением исследования были поставлены несколько гипотез:  
**Гипотеза 1**: Есть ли зависимость между количеством детей и возвратом кредита в срок?  
**Гипотеза 2**: Есть ли зависимость между семейным положением и возвратом кредита в срок?  
**Гипотеза 3**: Есть ли зависимость между уровнем дохода и возвратом кредита в срок?  
**Гипотеза 4**: Как разные цели кредита влияют на его возврат в срок?  

В результат удалось выявить зависимости:

**Гипотеза 1**: Частично подтверждена, для категории три и более детей необходимо провести дополнительный анализ, т.к. данных для анализа недостаточно.  
**Гипотеза 2**: Подтверждена, семейное положение оказывает влияние на процент успешной выплаты по кредиту.  
**Гипотеза 3**: Частично подтверждена, уровень дохода оказывает влияние на процент успешной выплаты по кредиту, для полноценного анализа необходимы дополнительные данные по категориям А, D, E.  
**Гипотеза 4**: Подтверждена, кредиты на операции с недвижимостью выплачиваются лучшем чем остальные, доля кредитов на проведение свадьбы не большая, для более точной аналитики требуется собрать больше данных.